### DISCUSSION TITLE
How are you identifying the dependencies between objects before deploying?

### DISCUSSION THREAD

**Discussion by prdpsvs** [Azure-Player/azure.datafactory.tools#102] [Q&A] [answered]
How do you know which objects are dependent on one another? I looked into function Deploy-AdfObject and see you are resolving dependencies but how are you doing it?

**Comment by NowinskiK** [comment, marked as answer]
Dependencies are stored in `DependsOn` property of each object, which is fulfilling in `Import-AdfObjects` cmdlet, line 29-32:
```
        $m = [regex]::matches($txt,'"referenceName":\s*?"(?<r>.+?)",[\n\r\s]+"type":\s*?"(?<t>.+?)"')
        $m | ForEach-Object {
            $o.AddDependant( $_.Groups['r'].Value, $_.Groups['t'].Value ) | Out-Null
        }
```

**Reply by prdpsvs** [reply]
How do you store the dependencies between two objects? Where do you store the dependencies? Could you explain with an example: Lets say, a dataset (ds_azuresql_table1) or linked service (ls_azure_sql) should be deployed first and pipeline (pl_copydata) in sequence.

**Reply by prdpsvs** [reply]
@NowinskiK , could you reply on the follow up question pls?

**Reply by NowinskiK** [reply]
@prdpsvs I already answered above.
If you run function `Import-AdfFromFolder` - it will return an instance (object) of [Adf class](https://github.com/SQLPlayer/azure.datafactory.tools/blob/master/private/Adf.class.ps1) that contains all objects in arrays plus some other properties. Everything is there.
![image](https://user-images.githubusercontent.com/25029325/118048229-e2a4e380-b373-11eb-8290-77d728d35581.png)

![image](https://user-images.githubusercontent.com/25029325/118048130-bb4e1680-b373-11eb-9a72-a2cdac0cbb51.png)
HTH

**Reply by prdpsvs** [reply]
Thank you, I missed what you were saying the first time. You are looking for referenceName property in datasets, pipelines and triggers json files and updating DependsOn in each object in $adf.
